src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php

```diff
@@ -118,7 +169,7 @@ private function joinRelations(QueryBuilder $queryBuilder, string $resourceClass
 
             $relationAlias = $relationAlias.++$j;
 
-            $this->joinRelations($queryBuilder, $mapping['targetEntity'], $propertyMetadataOptions, $associationAlias, $relationAlias, $method === 'leftJoin');
+            $this->joinRelations($queryBuilder, $mapping['targetEntity'], $propertyMetadataOptions, $associationAlias, $relationAlias, $method === 'leftJoin', ++$depth);
```

Comment:
I changed the way it's counted to count the number of joins. I reduced the default value to 30.